extension/src/ballistics/bullet/model.rs

```diff
@@ -0,0 +1,216 @@
+use std::f64::consts::PI;
+
+use rand::{distributions::Uniform, prelude::Distribution};
+use rand_chacha::ChaCha8Rng;
+
+use crate::{
+    ballistics::{
+        atmosphere::{
+            calculate_air_density, calculate_atmospheric_correction, calculate_roughness_length,
+            speed_of_sound, AtmosphereModel,
+        },
+        drag::{calculate_retard, DragFunction},
+        map::Map,
+    },
+    common::{Temperature, Vector3, GRAVITY_ACCEL},
+};
+
+const STD_AIR_DENSITY_ICAO: f64 = 1.22498;
+
+pub struct Bullet {
+    pub air_friction: f64,
+    pub bullet_caliber: f64,
+    pub bullet_length: f64,
+    pub bullet_mass: f64,
+    pub ballistic_coefficients: Vec<f64>,
+    pub velocity_boundaries: Vec<f64>,
+    pub atmosphere_model: AtmosphereModel,
+    pub drag_function: DragFunction,
+    pub muzzle_velocities: Vec<f64>,
+    pub barrel_length: f64,
+    pub stability_factor: f64,
+    pub twist_direction: f64,
+    pub transonic_stability_coefficient: f64,
+    pub bullet_velocity_last_frame: Vector3,
+    pub origin: Vector3,
+    pub latitude: f64,
+    pub temperature: Temperature,
+    pub altitude: f64,
+    pub humidity: f64,
+    pub overcast: f64,
+    pub start_time: f64,
+    pub last_frame: f64,
+    pub rng: ChaCha8Rng,
+}
+
+impl Bullet {
+    pub fn simulate(
+        &mut self,
+        map: &Map,
+        bullet_velocity_current_frame: Vector3,
+        bullet_position: Vector3,
+        mut wind: Vector3,
+        height_atl: f64,
+        tick_time: f64,
+    ) -> Vector3 {
+        const EARTH_ANGULAR_SPEED: f64 = 0.000_072_92;
+        let mut tof = tick_time - self.start_time;
+        let delta_time = tick_time - self.last_frame;
+
+        let temperature =
+            Temperature::new_celsius(self.temperature.as_celsius() - 0.0065 * bullet_position.z());
+        let pressure = (1010.32 - 10.0 * self.overcast)
```

Comment:
[c++ src](https://github.com/acemod/ACE3/blob/master/extensions/advanced_ballistics/AdvancedBallistics.cpp#L503)
```suggestion
        let pressure = (1013.25 - 10.0 * self.overcast)
```

Reply:
1013.25 is ICAO Standard Atmosphere Pressure in hPa. :+1: